src/optimizer/rule/regex_optimizations.cpp

```diff
@@ -12,17 +14,76 @@ namespace duckdb {
 RegexOptimizationRule::RegexOptimizationRule(ExpressionRewriter &rewriter) : Rule(rewriter) {
 	auto func = make_uniq<FunctionExpressionMatcher>();
 	func->function = make_uniq<SpecificFunctionMatcher>("regexp_matches");
-	func->policy = SetMatcher::Policy::ORDERED;
+	func->policy = SetMatcher::Policy::SOME;
```

Comment:
By changing this to `Some` instead of `Ordered` the order of the matches no longer matters, meaning we could have `[Constant, Expression]` instead of `[Expression, Constant]`. This likely causes some issues with the optimizer below. Maybe we should introduce a `PARTIAL_ORDERED` to accommodate the case here?